src/s3path.jl

```diff
@@ -1,14 +1,31 @@
-struct S3Path <: AbstractPath
+struct S3Path{A<:AbstractAWSConfig} <: AbstractPath
     segments::Tuple{Vararg{String}}
     root::String
     drive::String
     isdirectory::Bool
-    config::AWSConfig
+    config::A
+
+    # this method ensures that there is proper conversion to Tuple{Vararg{String}} for first arg,
+    # which can be a little tempremental if you don't explicitly handle it
+    function S3Path{A}(
+        segments,
+        root::AbstractString,
+        drive::AbstractString,
+        isdirectory::Bool,
+        config::A
+    ) where {A<:AbstractAWSConfig}
+        new{typeof(config)}(tuple((String(s) for s in segments)...), root, drive, isdirectory, config)
```

Comment:
I don't love that we need to do this iteration, conversion and splatting every time we construct the path type? Can we have an outer constructor that addresses the specific case where this is necessary (e.g., segments aren't `Tuple{Vararg{String}}`?

Reply:
I was in the process of making a simple MWE to show you why it so easily fails, but to my surprise, in my simple example I was not having the issues I had with this that prompted me to make this change.  I'll experiment with this a bit and see if I can come up with a better solution.

Reply:
FWIW, if you find the failure case it might be worth adding a test here and opening an issue against FilePathsBase.jl? Even if we can't fix the problem there generically, we could at least fix the system paths and include it in the test suite.